graphql/pod_resolver.go

```diff
@@ -0,0 +1,62 @@
+package graphql
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/utility"
+)
+
+// Type is the resolver for the type field.
+func (r *podResolver) Type(ctx context.Context, obj *model.APIPod) (string, error) {
+	return string(obj.Type), nil
+}
+
+// Status is the resolver for the status field.
+func (r *podResolver) Status(ctx context.Context, obj *model.APIPod) (string, error) {
+	return string(obj.Status), nil
+}
+
+// Task is the resolver for the task field.
+func (r *podResolver) Task(ctx context.Context, obj *model.APIPod) (*model.APITask, error) {
+	task, err := task.FindByIdExecution(utility.FromStringPtr(obj.TaskRuntimeInfo.RunningTaskID), obj.TaskRuntimeInfo.RunningTaskExecution)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding task %s: %s", utility.FromStringPtr(obj.TaskRuntimeInfo.RunningTaskID), err.Error()))
+	}
+	if task == nil {
+		return nil, nil
+	}
+	apiTask := &model.APITask{}
+	if err = apiTask.BuildFromService(task, &model.APITaskArgs{
```

Comment:
nit : preference for breaking this out of the if statement; that syntax is really only preferred for easy one liners.